src/SourceBrowser/src/BinLogToSln/Program.cs

```diff
@@ -14,8 +14,115 @@
 
 namespace BinLogToSln
 {
-    class Program
-    {
+    class Program
+    {
+        private static CompilerInvocation SelectBestInvocation(IGrouping<string, CompilerInvocation> invocationGroup)
+        {
+            var invocations = invocationGroup.ToList();
+            if (invocations.Count == 1)
+            {
+                return invocations[0];
+            }
+
+            Console.WriteLine($"Selecting best invocation for assembly '{invocationGroup.Key}' from {invocations.Count} candidates:");
+            foreach (var inv in invocations)
+            {
+                Console.WriteLine($"  - {inv.ProjectFilePath}");
+            }
+
+            // Score each invocation based on our criteria
+            var scoredInvocations = invocations.Select(inv => new
+            {
+                Invocation = inv,
+                Score = CalculateInvocationScore(inv)
+            }).ToList();
+
+            // Select the highest scored invocation
+            var best = scoredInvocations.OrderByDescending(x => x.Score).First();
+            
+            Console.WriteLine($"Selected: {best.Invocation.ProjectFilePath} (score: {best.Score})");
+            return best.Invocation;
+        }
+
+        private static int CalculateInvocationScore(CompilerInvocation invocation)
+        {
+            int score = 0;
+
+            try
+            {
+                // Prefer invocations with non-*.notsupported.cs source files
+                var sourceFiles = invocation.Parsed?.SourceFiles;
+                if (sourceFiles.HasValue)
+                {
+                    int totalSourceFiles = sourceFiles.Value.Length;
+                    int notSupportedFiles = sourceFiles.Value.Count(sf => sf.Path.Contains(".notsupported.cs"));
+                    
+                    if (totalSourceFiles > 0)
+                    {
+                        // Higher score for projects with fewer notsupported files relative to total
+                        score += (totalSourceFiles - notSupportedFiles) * 100;
+                        
+                        // Bonus for having no notsupported files at all
+                        if (notSupportedFiles == 0)
+                        {
+                            score += 1000;
+                        }
+                    }
+                }
```

Comment:
Let's not use this convention.  Instead check for a project property, let's call it `IsPlatformNotSupportedAssembly`